### PR TITLE
hheretic: init at 0.2.3

### PR DESCRIPTION
**Pull request by majiru**
###### Description of changes

This added 'hacked heretic' an open source linux port of the doom clone Heretic released by raven software. This is maintained by the same folks that handled the 'hacked hexen' so I figured I would take a look at that as well. So I did some cleanup and enabled sdl_mixer so that the midi soundtrack works as intended (if setup correctly).

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/hheretic/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    owner = "sezero";
+    repo = "hheretic";
+    rev = "hheretic-${finalAttrs.version}";
+    hash = "sha256-e9N869W8STZdLUBSscxEnF2Z+SrdVv8ARDL8AMe1SJ8=";
+  };
+
+  nativeBuildInputs = [ autoconf automake SDL.dev ];
+  buildInputs = [ SDL SDL_mixer ];
+  strictDeps = true;
+  enableParallelBuilding = true;
```
Comment (AndersonTorres) [inline pkgs/games/hheretic/default.nix:25]:
```suggestion
  nativeBuildInputs = [ 
    autoreconfHook 
    SDL.dev
  ];
  
  buildInputs = [
    SDL 
    SDL_mixer 
  ];
  
  strictDeps = true;
```

File: pkgs/games/hheretic/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  configurePhase = ''
+    runHook preConfigure
+
+    bash autogen.sh
+    ./configure --with-audio=sdlmixer
+
+    runHook postConfigure
+  '';
```
Comment (AndersonTorres) [inline pkgs/games/hheretic/default.nix:34]:
```suggestion
    configureFlags = [ "--with-audio=sdlmixer" ];
```

File: pkgs/games/hheretic/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "hheretic-";
+  };
+
+  meta = with lib; {
+    description = "Linux port of Raven Game's Heretic";
+    homepage = "https://hhexen.sourceforge.net/hhexen.html";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ moody ];
+  };
```
Comment (AndersonTorres) [inline pkgs/games/hheretic/default.nix:53]:
```suggestion
  meta = {
    description = "Linux port of Raven Game's Heretic";
    homepage = "https://hhexen.sourceforge.net/hhexen.html";
    license = lib.licenses.gpl2Plus;
    maintainers = with lib.maintainers; [ moody ];
    inherit (SDL.meta) platforms;
  };
```

File: pkgs/games/hhexen/default.nix [hunk 4]
```diff
@@ -1,23 +1,55 @@
-{ lib, fetchurl, SDL, stdenv }:
+{ lib
+, fetchFromGitHub
+, SDL
+, SDL_mixer
+, automake
```
Comment (AndersonTorres) [inline pkgs/games/hhexen/default.nix:5]: Same as before

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/games/hheretic/default.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    SDL.dev
+  ];
+
+  buildInputs = [
+    SDL
+    SDL_mixer
+  ];
+
+  strictDeps = true;
+  enableParallelBuilding = true;
```
Comment (AndersonTorres) [inline pkgs/games/hheretic/default.nix:32]: Is this needed? Autoconf usually enables paralellism.

Comment (majiru) [reply pkgs/games/hheretic/default.nix:32]: Yes it doesn't seem to build with multiple cores unless I specifically specify this, just double checked on my machine.

**Pull request merged**